### PR TITLE
clear compiler warnings

### PR DESCRIPTION
**Pull request by zgbset**

### CODE REVIEW

File: components/net/at_device/wifi_esp8266/esp8266.c [hunk 1]
```diff
@@ -211,6 +211,8 @@ int32_t esp8266_data_handler(void *arg, int8_t *buf, int32_t len)
     char *p1, *p2;
     QUEUE_BUFF qbuf;
     p1 = (char *)buf;
+    
+    (void)ret; // For keil unused warning
```
Comment (jimscen-wang) [inline components/net/at_device/wifi_esp8266/esp8266.c]: #define AT_LOG(...) (void)(__VA_ARGS__)
to suppress the unused warning course by  AT_LOG when no log enable

File: demos/nbiot_without_atiny/nb_demo.c [hunk 2]
```diff
@@ -78,7 +78,7 @@ void demo_nbiot_only(void)
         los_nb_report("22", 2);
         LOS_TaskDelay(60000);
     }
-    los_nb_report("23", 1);
+    //los_nb_report("23", 1);
```
Comment (jimscen-wang) [inline demos/nbiot_without_atiny/nb_demo.c]: why not report?

File: tests/test_suit/fs_test.c [hunk 3]
```diff
@@ -93,7 +93,7 @@ static void print_dir(const char *name, int level)
             && strcmp(dirent->name, ".")
             && strcmp(dirent->name, ".."))
         {
-            char tmp_path[128];
+            char tmp_path[LOS_MAX_DIR_NAME_LEN+2];
```
Comment (jimscen-wang) [inline tests/test_suit/fs_test.c]: why addend is 2 ant not 1?

Comment (zgbset) [reply tests/test_suit/fs_test.c]: char tmp_path[LOS_MAX_DIR_NAME_LEN+1];
snprintf(tmp_path, sizeof(tmp_path), "%s/%s", name, dirent->name);

#define LOS_MAX_DIR_NAME_LEN 255

struct dirent
{
    char                 name [LOS_MAX_DIR_NAME_LEN+1];
    UINT32               type;
    UINT32               size;
};

gcc warning: 'snprintf' output 2 or more bytes (assuming 257) into a destination of size 256

File: demos/nbiot_without_atiny/nb_demo.c (hunk 2 above)
Comment (zgbset) [reply demos/nbiot_without_atiny/nb_demo.c]: > why not report?

while(1)
{
	los_nb_report("22", 2);
	LOS_TaskDelay(60000);
}
los_nb_report("23", 1);

..\..\..\demos\nbiot_without_atiny\nb_demo.c(81): warning:  #111-D: statement is unreachable

**Pull request merged**